man: move filesystems to section 4

concussious opened the pull request:
This was originally inspired by some cleanups and clarifications for msdosfs, which will be saved for another PR.

igalic commented on share/man/man5/msdosfs.5:
I would remove this in the previous commit, this way the rename is atomic.

concussious replied:
Thank you so much for review. Sorry, I didn't understand "git rm" when i started. I tried to rebase but it seemed like I could only squash a commit to the directly previous commit. Is there a way to do this without starting over?

igalic replied:
```
# uncommit the last commit
git reset --soft HEAD^
# commit the "deletion into the rename" 
git add share/man/man5/msdosfs.5
git commit --ammend share/man/man5/msdosfs.5
# now recreate the second commit
git add share/man/man4/msdosfs.
git commit
# the commit message will be lost in your clone, but you can still copy paste it from Github
git push -f
```

concussious replied:
Thank you so much

igalic replied:
thank you for doing the actual work!

igalic reviewed (changes requested):
one comment only so far

concussious commented:
> Note there are many filesystems in section 5; we don't want to treat msdos.5 differently from the others.

Agree. Please excuse me, I didn't see those.

> I'm not sure if it's better to rename section 5 to include filesystems or move all of them.

I believe my perspective, coming from a background without formal education but consistently using BSD since early childhood, is valuable for this discussion. I find it elegant and logical to group drivers together. As a humble handyman and taiji coach, I wondered, 'How did I never notice this driver page in Section 5? It should be with the other drivers.' Upon closer inspection of intro(4), it only specifies 'device' drivers.

If there's a consensus, I'd be honored to move the rest to Section 4 and make adjustments to intro.4 and man.1 accordingly.

Maybe this kind of change would be better discussed on a differential? I created an account some months ago, but I haven't quite figured out how to use it yet.

bsdimp commented:
> Note there are many filesystems in section 5; we don't want to treat msdos.5 differently from the others. I'm not sure if it's better to rename section 5 to include filesystems or move all of them.
> 
> cd9660.5 ext2fs.5 fusefs.5 linprocfs.5 linsysfs.5 mqueuefs.5 msdosfs.5 nullfs.5 smbfs.5 tarfs.5 tmpfs.5 unionfs.5

I kinda think we should move these. Section 5 is file formats, and most of the file format are there. All these documents are not about the format of these filesysttems, but how to configure them for the kernel.

But one thing I noticed when I was staging this... msdosfs.5 wasn't remove from share/man/man5/Makefile, nor msdosfs.4 added to share/man/man4/Makefile. That would need to be corrected. And you should do it in the first.

Neither NetBSD nor OpenBSD have these in man5. In fact, the options are described elsewhere that I'm having trouble finding.

concussious commented:
> But one thing I noticed when I was staging this... msdosfs.5 wasn't remove from share/man/man5/Makefile, nor msdosfs.4 added to share/man/man4/Makefile. That would need to be corrected. And you should do it in the first.

Oops, thank you. Fixed in latest push.

bsdimp commented on sbin/fsck_ffs/fsck_ffs.8:
```diff
@@ -31,7 +33,8 @@
 .Nm fsck_ffs ,
-.Nm fsck_ufs
+.Nm fsck_ufs ,
+.Nm fsck_4.2bsd
```
This looks to be a separate change...  Can you split it out into its own commit?

bsdimp commented:
Can you split out the changes that aren't directly related to the move into a separate commit (I tagged one, then saw a lot then saw your commit message)? Most of them don't look like they change more than markup, so it's good you've not bumped the dates (which can be a real pain to merge).
I think after that, it will be ready.

concussious commented:
I didn't change any of the docdates this time. I'd like to add something to style.mdoc about not bumping docdates if it makes it harder for committers.

Should contributors always save bumping docdate for the committer?

bsdimp commented:
> Should contributors always save bumping docdate for the committer?

I think so on github, though that's poorly documented. And we don't have a nice script that will just do it for everything in the commit. This time, though, ir does make sense.

I think this is ready, but I want to do a final look over in my staging area.

concussious commented:
I'm sorry, looking over it again: the "CD" I added to cd9660.4 is unhelpful because CD is already a valid search term from Nm; so I switched it to "disc", because it seems reasonable to search for disc sometimes.

My proposed addition of "driver" to msdosfs.4 isn't appropriate or consistent, if it's being moved to (4), a 4 will appear in the listing denoting it's drivership.

bsdimp commented:
Hmmm, since we're moving all these files, I think we need ObsoleteFiles.inc changes, no? Sorry I didn't think of it sooner.

concussious commented:
> we need ObsoleteFiles.inc changes

I think I also need to grep all pages in the manual for xrefs to these pages and fix them, right?

bsdimp commented:
> > we need ObsoleteFiles.inc changes
> 
> I think I also need to grep all pages in the manual for xrefs to these pages and fix them, right?

Yes please.

concussious commented:
Sorry, I understand this is unreviewable and I need to break it up more, but I wanted to get what I had up for now.

dag-erling commented:
Can you please narrow this PR to only what it says it does?

concussious commented:
After reflecting on your example this morning, I decided to narrow it even further to just the move. Thanks.

bsdimp commented:
Are you still working on this, or were the pushes just the moves... I didn't see anything stray, but I wanted to check

dag-erling commented on lib/libsys/statfs.2:
```diff
@@ -106,8 +106,8 @@ All I/O to the file system is done synchronously.
-Soft updates being done (see
-.Xr ffs 7 ) .
+Soft updates being done
+.Pq see Xr ffs 4 .
```
Please leave the markup the way it was. If you absolutely want to change it (and the approximately 600 other occurrences) you can do it in a separate PR and use `Po` / `Pc`.

dag-erling replied:
I'm not kidding about 600 btw:

````
% git grep '\(see$' '*.[0-9]' | wc -l
     590
````

dag-erling reviewed (changes requested):
Excellent work with just one tiny caveat.

concussious commented:
> Are you still working on this

I believe it's ready. Thank you for your patience.
